src/main/java/net/snowflake/ingest/streaming/internal/SnowflakeStreamingIngestClientInternal.java

```diff
@@ -499,56 +447,36 @@ public Map<String, String> getLatestCommittedOffsetTokens(
-
-      ChannelsStatusResponse response =
-          executeWithRetries(
-              ChannelsStatusResponse.class,
-              CHANNEL_STATUS_ENDPOINT,
-              payload,
-              "channel status",
-              STREAMING_CHANNEL_STATUS,
-              httpClient,
-              requestBuilder);
-
-      // Check for Snowflake specific response code
-      if (response.getStatusCode() != RESPONSE_SUCCESS) {
-        throw new SFException(ErrorCode.CHANNEL_STATUS_FAILURE, response.getMessage());
-      }
+    ChannelsStatusRequest request = new ChannelsStatusRequest();
+    List<ChannelsStatusRequest.ChannelStatusRequestDTO> requestDTOs =
+        channels.stream()
+            .map(ChannelsStatusRequest.ChannelStatusRequestDTO::new)
+            .collect(Collectors.toList());
+    request.setChannels(requestDTOs);
+    request.setRole(this.role);
+    request.setRequestId(this.storageManager.getClientPrefix() + "_" + counter.getAndIncrement());
 
-      for (int idx = 0; idx < channels.size(); idx++) {
-        SnowflakeStreamingIngestChannelInternal<?> channel = channels.get(idx);
-        ChannelsStatusResponse.ChannelStatusResponseDTO channelStatus =
-            response.getChannels().get(idx);
-        if (channelStatus.getStatusCode() != RESPONSE_SUCCESS) {
-          String errorMessage =
-              String.format(
-                  "Channel has failure status_code, name=%s, channel_sequencer=%d, status_code=%d",
-                  channel.getFullyQualifiedName(),
-                  channel.getChannelSequencer(),
-                  channelStatus.getStatusCode());
-          logger.logWarn(errorMessage);
-          if (getTelemetryService() != null) {
-            getTelemetryService()
-                .reportClientFailure(this.getClass().getSimpleName(), errorMessage);
-          }
+    ChannelsStatusResponse response = snowflakeServiceClient.channelStatus(request);
+
+    for (int idx = 0; idx < channels.size(); idx++) {
+      SnowflakeStreamingIngestChannelInternal<?> channel = channels.get(idx);
+      ChannelsStatusResponse.ChannelStatusResponseDTO channelStatus =
+          response.getChannels().get(idx);
+      if (channelStatus.getStatusCode() != RESPONSE_SUCCESS) {
+        String errorMessage =
+            String.format(
+                "Channel has failure status_code, name=%s, channel_sequencer=%d, status_code=%d",
+                channel.getFullyQualifiedName(),
+                channel.getChannelSequencer(),
+                channelStatus.getStatusCode());
+        logger.logWarn(errorMessage);
+        if (getTelemetryService() != null) {
+          getTelemetryService().reportClientFailure(this.getClass().getSimpleName(), errorMessage);
         }
       }
-
-      return response;
-    } catch (IOException | IngestResponseException e) {
-      throw new SFException(e, ErrorCode.CHANNEL_STATUS_FAILURE, e.getMessage());
     }
+
+    return response;
   }
```

Comment:
lets bring this catch block back, as it was covering the reportClientFailure() call too.